src/components/Form.js

```diff
@@ -1,15 +1,15 @@
 import {nanoid} from 'nanoid';
 import {Button} from './Button';
 
-const Form = ({input, setInput, todos, setTodos}) => {
+const Form = ({input, newInput, todos, newTodo}) => {
   const onInputChange = e => {
-    setInput(e.target.value);
+    newInput(e.target.value);
```

Comment:
The whole logic should be at the level your state is managed. Keep your components stupid as possible.